BL-352 - hero edit permisions

randyhoulahan opened the pull request.

randyhoulahan committed:
BL-352 fix bread crumbs

sbilodeau commented on stores/menus.js:
```diff
@@ -54,57 +81,69 @@ export const useMenusStore = defineStore('menus', {
+            if(!id) return false;
+            if(!this?.main?.length) return false;
+
+            for(let i = 0; i < this.main.length; i++)
+                if(this.main[i].children?.length)
+                    for(let j = 0; j < this.main[i].children.length; j++)
+                        if(this.main[i].children[j].length)
+                            if(this.isInMenuByContentTypeId(this.main[i].children[j], id)) 
+                                return this.isInMenuByContentTypeId(this.main[i].children[j], id)
```
why not use `for... of`
More or less
```suggestion
for(let entry of this.main.length || [])
    for(let child of entry.children || [])
        if(this.isInMenuByContentTypeId(child, id)) 
            return this.isInMenuByContentTypeId(child, id)
```

randyhoulahan replied:
applied thanks

sbilodeau commented on stores/menus.js:
```diff
@@ -54,57 +81,69 @@ export const useMenusStore = defineStore('menus', {
             const term = alias? this.getSystemPageByAlias(alias, locale) : this.getSystemPageById(id);
 
-            if(!term) return `/taxonomy/term/21`;
+            if(!term || !term?.aliases?.length) return id? `/taxonomy/term/${id}`:`/taxonomy/term/${systemPageTidConstants.SEARCH}`;
```
no encoding?

randyhoulahan replied:
fixed https://github.com/scbd/bioland-head/commit/bf0c297

randyhoulahan committed:
Update stores/menus.js

Co-authored-by: Stephane Bilodeau <bilodeaustephane@gmail.com>

randyhoulahan committed:
BL-352: Add encodeURIComponent to URL path construction

Added encodeURIComponent() to the id and systemPageTidConstants.SEARCH
values in getSystemPagePath() to properly encode special characters in
taxonomy term URLs.

Addresses review comment: https://github.com/scbd/bioland-head/pull/46#discussion_r2033767679

randyhoulahan committed:
BL-352: Refactor for loops to for...of in menus store

- Replace traditional for loops with for...of loops for improved readability
- Updated methods: isInMenu, isInMenuId, isInMainMenu, isInMainMenuByContentTypeId, isInMenuByContentTypeId
- Use descriptive variable names (child, grandChild, menuItem) instead of index-based access

randyhoulahan committed:
Merge branch 'master' into BL-352

rqk2k commented on components/page/bread-crumbs.vue:
```diff
@@ -101,6 +103,7 @@
         else{
 
             return  menusStore.isInMainMenuByContentTypeId(contentTypeId.value)?.crumbs || [];
+            // if(!inMenu.value) return [];
```
Isn't it better to just remove it instead of commenting?
